tools/infer/utility.py

```diff
@@ -293,6 +299,28 @@ def create_predictor(args, mode, logger):
             config.enable_custom_device("mlu")
         elif args.use_xpu:
             config.enable_xpu(10 * 1024 * 1024)
+        elif args.use_gcu:  # for Enflame GCU(General Compute Unit)
+            import paddle_custom_device.gcu.passes as gcu_passes
+
+            gcu_passes.setUp()
+            if args.precision == "fp16":
+                config.enable_custom_device(
+                    "gcu", 0, paddle.inference.PrecisionType.Half
+                )
+                gcu_passes.set_exp_enable_mixed_precision_ops(config)
+            else:
+                config.enable_custom_device("gcu")
+
+            if paddle.framework.use_pir_api():
+                config.enable_new_ir(True)
+                config.enable_new_executor(True)
+                kPirGcuPasses = gcu_passes.inference_passes(
```

Comment:
Why do we need to build passes based on "PaddleOCR" or "PaddleXXX"?

Reply:
We need to optimize performance based on hardware features and even network structure, and the processing here is to increase this flexibility.